ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -394,29 +394,29 @@ private[apiserver] final class ApiUserManagementService(
       rights: Set[UserRight],
       identityProviderId: IdentityProviderId,
       isParticipantAdmin: Boolean,
-  ): Future[Unit] =
-    if (isParticipantAdmin) Future.unit
-    else partyExistsOrError(userParties(rights), identityProviderId)
-
-  private def partyExistsOrError(
-      parties: Set[Ref.Party],
-      identityProviderId: IdentityProviderId,
-  ): Future[Unit] =
-    partyRecordExist
-      .filterPartiesExistingInPartyRecordStore(identityProviderId, parties)
+  ): Future[Unit] = {
+    val parties = userParties(rights)
+    val partiesExistingInPartyRecordStore =
+      if (isParticipantAdmin) partyRecordExist.filterPartiesExistingInPartyRecordStore(parties)
+      else
+        partyRecordExist
+          .filterPartiesExistingInPartyRecordStore(identityProviderId, parties)
+
+    partiesExistingInPartyRecordStore
       .flatMap { partiesExist =>
         val partiesNotExist = parties -- partiesExist
         if (partiesNotExist.isEmpty)
-          Future.successful(())
+          Future.unit
         else
           Future.failed(
             LedgerApiErrors.RequestValidation.InvalidArgument
               .Reject(
-                s"Provided parties [${partiesNotExist.mkString(",")}] have not been found in identity_provider_id=${identityProviderId.toRequestString}."
+                s"Provided parties [${partiesNotExist.mkString(",")}] have not been found in identity_provider_id=`${identityProviderId.toRequestString}`."
```

Comment:
I guess `partiesNotExist.mkString(",")` should be truncated not to be too long.
Otherwise default error codes truncation takes place and this suffix `have not been found in identity_provider_id=`${identityProviderId.toRequestString}` might get truncated